xml-reader/src/main/java/org/triplea/generic/xml/reader/AnnotatedFields.java

```diff
@@ -0,0 +1,131 @@
+package org.triplea.generic.xml.reader;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import lombok.Getter;
+import org.triplea.generic.xml.reader.annotations.Attribute;
+import org.triplea.generic.xml.reader.annotations.BodyText;
+import org.triplea.generic.xml.reader.annotations.Tag;
+import org.triplea.generic.xml.reader.annotations.TagList;
+import org.triplea.generic.xml.reader.exceptions.JavaDataModelException;
+
+/**
+ * Sorts fields, validates fields, and sets fields to accessible. Field validation will check that
+ * we see the right annotations on the right class types, for example that '@TagList' is on only
+ * 'java.util.List'.
+ *
+ * @param <T> Java model object class type (essentially the tag we are currently creating as a java
+ *     object).
+ */
+@Getter
+class AnnotatedFields<T> {
+  private final List<Field> attributeFields = new ArrayList<>();
+  private final List<Field> tagFields = new ArrayList<>();
+  private final List<Field> tagListFields = new ArrayList<>();
+  private final List<Field> bodyTextFields = new ArrayList<>();
+
+  AnnotatedFields(final Class<T> pojo) throws JavaDataModelException {
+    for (final Field field : pojo.getDeclaredFields()) {
+      validateAnnotations(field);
+      if (field.getAnnotation(Attribute.class) != null) {
+        field.setAccessible(true);
+        attributeFields.add(field);
+      } else if (field.getAnnotation(Tag.class) != null) {
+        field.setAccessible(true);
+        tagFields.add(field);
+      } else if (field.getAnnotation(TagList.class) != null) {
+        field.setAccessible(true);
+        tagListFields.add(field);
+      } else if (field.getAnnotation(BodyText.class) != null) {
+        field.setAccessible(true);
+        bodyTextFields.add(field);
+      }
+    }
+    if (bodyTextFields.size() > 1) {
+      throw new JavaDataModelException(
+          "Too many body text fields, can only have one on any given class");
+    }
+    if (!bodyTextFields.isEmpty() && (!tagFields.isEmpty() && !tagListFields.isEmpty())) {
+      throw new JavaDataModelException(
+          "Illegal combination of annoations, may only have attributes and a body text,"
+              + "or attributes and tags (or taglist), but may not have both body text and tags.");
+    }
+  }
+
+  private static <T> void validateAnnotations(final Field field) throws JavaDataModelException {
```

Comment:
Codacy found an issue: [The method 'validateAnnotations(Field)' has an NPath complexity of 10206, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6130586)